File: C++/Constructors.cpp
class aClass {
public:
	aClass(): _a(1) {}
	aClass(int a): _a(a) {}
	~aClass() {_a = 0;}
private:
	int _a;
};

int main(int argc, char const *argv[]) {
	aClass a1; //calls the default constructor
	aClass a2(3);
	return 0;
}

File: objC/Constructors.m
#import <Foundation/Foundation.h>

@interface aClass: NSObject {
	int a;
}
- (id) initWithA: (int) aValue;
@end

@implementation aClass

- (id) init {
	if (self = [super init]) {
		a = 1;
	}
	return self;
}

- (id) initWithA: (int) aValue {
	if (self = [self init]) {
		a = aValue;
	}
	return self;
}

- (void) dealloc {
	[super dealloc];
	a = 0;
}

@end

int main(int argc, char const *argv[])
{
	aClass *a1 = [[aClass alloc] init];
	aClass *a2 = [[aClass alloc] initWithA: 5];
	return 0;
}